src/DecayChannel.cxx

```diff
@@ -177,6 +178,20 @@ void DecayChannel::addParticleCombination(std::shared_ptr<ParticleCombination> p
 }
 
 //-------------------------
+void DecayChannel::registerWithModel()
+{
+    for (auto& d : Daughters_) {
+        auto dp = std::dynamic_pointer_cast<DecayingParticle>(d);
+        if (dp)
+            dp->registerWithModel();
```

Comment:
In favor of not having to dynamic_cast---add pure virtual function `Particle::registerWithModel()`, and override in `DecayingParticle` (and `FinalStateParticle`).